rfc/scope_manager.md

```diff
@@ -0,0 +1,61 @@
+# Scope Manager
+
+**Current State:** Draft  
+**Author:** [carlosalberto](https://github.com/carlosalberto)
+
+In the OpenTracing specification, under the "Optional API Elements" section, it is mentioned languages may choose to provide utilities to pass an active **Span** around a single process.
+
+Upon many iterations and feedback from several contributors, the Java 0.31 API defined the new **Scope Manager** concept, which is a simple and explicit way to manage the active **Span** at a given call-context point. This document intends to standardize this concept, so other supported languages and platforms can leverage it, with their respective semantical differences.
+
+# Technical background
+
+For any thread, at most one **Span** may be "active". Of course there may be many other **Spans** involved with the thread which are (a) started, (b) not finished, and yet (c) not "active": perhaps they are waiting for I/O, blocked on a child **Span**, or otherwise off of the critical path.
+ 
+It's inconvenient to pass an active **Span** from function to function manually, so OpenTracing should require that every **Tracer** contains a **Scope Manager** that grants access to the active **Span** through a container, called **Scope**.
```

Comment:
I'd rewrite this as something like: "For applications where the whole unit of work happens in a single thread, it's more convenient to store the active span in a thread-local variable than to pass the span context to all involved functions."

I wouldn't say that passing the context down the chain is "inconvenient" (it's why Go's context was made for, IIRC). Rather, it's *more* convenient to not pass this context around, whenever possible. Unfortunately, new generation applications using reactive/stream patterns can't make use of thread-local vars, so, passing down the context is the right approach.